repair/row_level.cc

```diff
@@ -676,6 +676,28 @@ void flush_rows(schema_ptr s, std::list<repair_row>& rows, lw_shared_ptr<repair_
     }
 }
 
+struct global_reader_permit_meta {
+    std::vector<foreign_ptr<lw_shared_ptr<semaphore_units<>>>> permits{smp::count};
+};
+
+future<global_reader_permit_meta> get_global_reader_permit(repair_service& rs, unsigned shard, bool all_shards) {
```

Comment:
If "global permit" means a permit for each of the shards, like I understood above, what is the "all_shards=false" supposed to mean? A comment above this method might have helped.

Reply:
all_shards = false means, a reader is going to take the semaphore only on a specific shard. For example, all local reader is going to take only the semaphore on local shard (single shard). All multi-shard reader are going to take the lock on all shards.

Reply:
But you called the function and the type a "global permit". Until now I understood that this means "a permit on all shards", then now I understand that we can actually have a "global permit with all_shards=false" which is... not so global... So I don't understand what exactly a "global permit" is.

Reply:
I could not think of a better name. I dropped the global permit name in v2.

Reply:
I think that whatever you call it, you need a comment, explaining what this object is and that in some cases it takes permits on just one shard and in other cases on all shards - and why it does this.

Reply:
There are comment in v2 too.